sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects.scala

```diff
@@ -359,10 +359,27 @@ case class LambdaVariable(value: String, isNull: String, dataType: DataType) ext
 case class MapObjects(
     function: AttributeReference => Expression,
     inputData: Expression,
-    elementType: DataType) extends Expression {
+    elementType: DataType)(
+    completeFunctionCopy: Option[Expression] = None) extends Expression {
```

Comment:
I know that during our offline discussion, I said this alternative sounded good, but later on I realized that this was equivalent to another solution I tried before, and it doesn't work because making `function: Expression` a constructor argument forces the analyzer to resolve `function` eagerly, while the `loopVar` inside `function` can't be resolved during analysis phase.